x-pack/plugins/security_solution/server/lib/telemetry/tasks/configuration.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/core/server';
+import type { ITelemetryEventsSender } from '../sender';
+import type { TelemetryConfiguration } from '../types';
+import type { ITelemetryReceiver } from '../receiver';
+import type { TaskExecutionPeriod } from '../task';
+import { artifactService } from '../artifact';
+import { telemetryConfiguration } from '../configuration';
+import { tlog } from '../helpers';
+
+export function createTelemetryConfigurationTaskConfig() {
+  return {
+    type: 'security:telemetry-configuration',
+    title: 'Security Solution Telemetry Configuration Task',
+    interval: '45m',
+    timeout: '1m',
+    version: '1.0.0',
+    runTask: async (
+      taskId: string,
+      logger: Logger,
+      receiver: ITelemetryReceiver,
+      sender: ITelemetryEventsSender,
+      taskExecutionPeriod: TaskExecutionPeriod
+    ) => {
+      try {
+        const artifactName = 'telemetry-configuration-v1';
+        const configArtifact = (await artifactService.getArtifact(
```

Comment:
I think I would approach this slightly differently so I wasn't updating a state every 45m. As the decoded hash is on the config:

```yml
"telemetry-filterlists-v1": {
  "decoded_sha256": "8dc364fe0c2f6cf9b0516cb00f5eeee874a575a8ae216803cd4ca6c43b3358a9",
  "decoded_size": 28152,
  "relative_url": "/downloads/kibana/telemetry-filterlists-v1/8dc364fe0c2f6cf9b0516cb00f5eeee874a575a8ae216803cd4ca6c43b3358a9"
}
```

Instead of returning `0` from this method I would return the decoded SHA256 and then check if the config needs updated.

Now that I have written the above, I think an even better approach is to send a HEAD request and if the e-tag is different to the one cached, download the new artifact, or skip the whole thing. That would also be a more courteous way of handling customer resources.